deepfence_frontend/apps/dashboard/src/features/auth/pages/ForgotPassword.tsx

```diff
@@ -1,18 +1,75 @@
 import cx from 'classnames';
-import { Link, useFetcher } from 'react-router-dom';
-import { Button, TextInput, Typography } from 'ui-components';
+import { ActionFunctionArgs, Link, useFetcher } from 'react-router-dom';
+import { toast } from 'sonner';
+import { Button, CircleSpinner, TextInput, Typography } from 'ui-components';
 
+import { getUserApiClient } from '@/api/api';
+import { ApiDocsBadRequestResponse } from '@/api/generated';
 import LogoDarkBlue from '@/assets/logo-deepfence-dark-blue.svg';
+import { ApiError, makeRequest } from '@/utils/api';
 
-export const forgotPasswordAction = async () => {
+export type actionReturnType = {
+  error?: string;
+  fieldErrors?: {
+    email?: string;
+  };
+  message?: string;
+  success?: boolean;
+};
+
+export const forgotPasswordAction = async ({
+  request,
+}: ActionFunctionArgs): Promise<actionReturnType> => {
+  const formData = await request.formData();
+  const body = Object.fromEntries(formData);
+  if (!body.email) {
+    return {
+      fieldErrors: {
+        email: 'Email is required',
+      },
+    };
+  }
+  const r = await makeRequest({
+    apiFunction: getUserApiClient().resetPasswordRequest,
+    apiArgs: [
+      {
+        modelPasswordResetRequest: {
+          email: body.email as string,
+        },
+      },
+    ],
+    errorHandler: async (r) => {
+      const error = new ApiError<actionReturnType>({});
+      if (r.status === 400) {
+        const modelResponse: ApiDocsBadRequestResponse = await r.json();
+        return error.set({
+          fieldErrors: {
+            email: modelResponse.error_fields?.email as string,
+          },
+        });
+      } else if (r.status === 403) {
+        const modelResponse: ApiDocsBadRequestResponse = await r.json();
+        return error.set({
+          error: modelResponse.message,
+        });
+      }
+    },
+  });
+
+  if (ApiError.isApiError(r)) {
+    return r.value();
+  }
+  toast.success(r.message);
```

Comment:
We don't want to toast on auth page, please add a success message below the button.

Reply:
fixed it